File: aizu online judge/練習/4章 データ構造/4.2_stack.py
"""
https://judge.u-aizu.ac.jp/onlinejudge/description.jsp?id=ALDS1_3_A&lang=ja
"""

s = []

for i in input().split():
    if i == "+":
        a = s.pop()
        b = s.pop()
        s.append(a + b)
    elif i == "-":
        a = s.pop()
        b = s.pop()
        s.append(b - a)
    elif i == "*":
        a = s.pop()
        b = s.pop()
        s.append(a * b)
    else:
        s.append(int(i))

print(s.pop())

File: aizu online judge/練習/4章 データ構造/c++/4.2 stack.cpp
// https://judge.u-aizu.ac.jp/onlinejudge/description.jsp?id=ALDS1_3_A&lang=ja

#include <iostream>
using namespace std;

int top, S[1000];

void push(int x) {
  S[++top] = x;
}

int pop() {
  top--;
  return S[top+1];
}

int main() {
  int a, b;
  top = 0;
  char s[100];

  while(cin >> s) {
    if (s[0] == '+') {
      a = pop();
      b = pop();
      push(a + b);
    } else if (s[0] == '-') {
      b = pop();
      a = pop();
      push(a - b);
    } else if (s[0] == '*') {
      a = pop();
      b = pop();
      push(a * b);
    } else {
      push(atoi(s));
    }
  }

  cout << pop() << endl;
  return 0;
}